cmd/query/app/http_handler.go

```diff
@@ -425,18 +431,38 @@ func (aH *APIHandler) parseTraceID(w http.ResponseWriter, r *http.Request) (mode
 	return traceID, true
 }
 
+func (aH *APIHandler) parseMicroseconds(w http.ResponseWriter, r *http.Request, timeKey string) (time.Time, bool) {
+	if timeString := r.FormValue(timeKey); timeString != "" {
+		t, err := aH.queryParser.parseTime(r, timeKey, time.Microsecond)
+		if aH.handleError(w, err, http.StatusBadRequest) {
+			return time.Time{}, false
+		}
+		return t, true
+	}
+	// It's OK if no time is specified, since it's optional
+	return time.Time{}, true
+}
+
+func (aH *APIHandler) parseGetTraceParameters(w http.ResponseWriter, r *http.Request) (spanstore.GetTraceParameters, bool) {
+	traceID, traceIdOk := aH.parseTraceID(w, r)
+	startTime, startTimeOk := aH.parseMicroseconds(w, r, startTimeParam)
```

Comment:
you cannot defer checking the error like that because helper functions already called `aH.handleError` which writes to the client (you cannot do that more than once).

Reply:
I see, updated as checking result immediately